shared/fs/container.js

```diff
@@ -1,16 +1,57 @@
 // @flow
+import * as I from 'immutable'
 import {connect, type TypedState, type Dispatch} from '../util/container'
 import * as FSGen from '../actions/fs-gen'
-import Fs from '.'
+import Files from '.'
+import {navigateAppend, navigateUp} from '../actions/route-tree'
+import * as Types from '../constants/types/fs'
 
-const mapStateToProps = (state: TypedState) => ({
-  counter: state.fs.counter,
-  you: state.config.username,
+type StateProps = {
+  you: ?string,
+  path: Types.Path,
+  items: I.List<string>,
+  pathItems: I.Map<Path, PathItem>,
+}
+
+type DispatchProps = {
+  onViewFolder: (path: Types.Path) => void,
+}
+
+const mapStateToProps = (state: TypedState, ownProps): StateProps => {
+  var path = state.fs.defaultPath
+  if (ownProps.routeProps) {
```

Comment:
you will always have routeProps unless you're embedding this else where (likely not happening). so better to just simplify and remove it